libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/camera/data/ViewportDataSourceProcessorTest.kt

```diff
@@ -0,0 +1,595 @@
+            every { legIndex } returns 1
+        }
+        val expected: List<Point> = listOf(
+            Point.fromLngLat(-77.168728, 38.777728),
+            Point.fromLngLat(-77.168619, 38.777744),
+            Point.fromLngLat(-77.168499, 38.777768),
+            Point.fromLngLat(-77.167878, 38.777921),
+            Point.fromLngLat(-77.1676211, 38.7779783)
+        )
+
+        val actual = getPointsToFrameAfterCurrentManeuver(
+            frameGeometryAfterManeuverEnabled = true,
+            generatedPostManeuverFramingPoints = postManeuverFramingPoints,
+            currentLegProgress = legProgress,
+            currentStepProgress = stepProgress
+        )
+
+        assertArrays1(expected, actual, pointAdapter)
+    }
+
+    @Test
+    fun `test getAllRemainingPointsOnRoute - empty current step`() {
+        val stepProgress: RouteStepProgress = mockk {
+            every { stepIndex } returns 4
+        }
+        val legProgress: RouteLegProgress = mockk {
+            every { legIndex } returns 1
+        }
+        val expected: List<Point> = listOf(
+            Point.fromLngLat(-77.153415, 38.771307),
+            Point.fromLngLat(-77.153451, 38.771105),
+            Point.fromLngLat(-77.153461, 38.770924),
+            Point.fromLngLat(-77.153468, 38.77091)
+        )
+
+        val actual = getAllRemainingPointsOnRoute(
+            completeRoutePoints = completeRoutePoints,
+            pointsToFrameOnCurrentStep = emptyList(),
+            currentLegProgress = legProgress,
+            currentStepProgress = stepProgress
+        )
+
+        assertArrays1(expected, actual, pointAdapter)
+    }
+
+    @Test
+    fun `test getAllRemainingPointsOnRoute`() {
+        val stepProgress: RouteStepProgress = mockk {
+            every { stepIndex } returns 4
+        }
+        val legProgress: RouteLegProgress = mockk {
+            every { legIndex } returns 1
+        }
+        val expected: List<Point> = listOf(
+            Point.fromLngLat(-77.123, 38.77091),
+            Point.fromLngLat(-77.456, 38.77091),
+            Point.fromLngLat(-77.153415, 38.771307),
+            Point.fromLngLat(-77.153451, 38.771105),
+            Point.fromLngLat(-77.153461, 38.770924),
+            Point.fromLngLat(-77.153468, 38.77091)
+        )
+
+        val actual = getAllRemainingPointsOnRoute(
+            completeRoutePoints = completeRoutePoints,
+            pointsToFrameOnCurrentStep = listOf(
+                Point.fromLngLat(-77.123, 38.77091),
+                Point.fromLngLat(-77.456, 38.77091)
+            ),
+            currentLegProgress = legProgress,
+            currentStepProgress = stepProgress
+        )
+
+        assertArrays1(expected, actual, pointAdapter)
+    }
+
+    @Test
+    fun getMapAnchoredPaddingFromUserPadding() {
+    }
+
+    @Test
+    fun getScreenBoxForFraming() {
+    }
+
+    @Test
+    fun getSmootherBearingForMap() {
+    }
+
+    @Test
+    fun normalizeBearing() {
+    }
```

Comment:
Should we remove and add as `TODO` / cut a ticket to follow up instead?

Reply:
Added all of them.